test-runner/src/logging.rs

```diff
@@ -53,3 +58,91 @@ impl log::Log for StdOutBuffer {
 pub fn init_logger() -> Result<(), SetLoggerError> {
     log::set_logger(&*LOGGER).map(|()| log::set_max_level(LevelFilter::Info))
 }
+
+pub async fn get_mullvad_app_logs() -> LogOutput {
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    let settings_json = {
+        const SETTINGS_PATH: &str = "/etc/mullvad-vpn/";
+        let settings_json_path = Path::new(SETTINGS_PATH).join("settings.json");
+        read_to_string(&settings_json_path).await.map_err(|e| {
+            Error::Logs(format!(
+                "{}: {}",
+                settings_json_path.to_str().unwrap_or("unknown path"),
+                e
+            ))
+        })
+    };
+    #[cfg(target_os = "windows")]
+    let settings_json = {
+        mullvad_paths::get_default_settings_dir()
+            .and_then(|mut path| {
+                path.push("settings.json");
+                read_to_string(&path).map_err(|e| {
```

Comment:
Isn't this missing an `await` somewhere? As well, there's no reason to repeat the call to `read_to_string`. Just making the target-specific code obtain the path would mean less duplication.

Reply:
Yes, fixed this by using the paths crate (that we figured out works for all OS:es).